pkg/controllers/manager.go

```diff
@@ -17,7 +17,8 @@ import (
 )
 
 const (
-	WorkerAddr = "status.tasks.workeraddr"
+	WorkerAddrIndex       = ".status.tasks.workerAddr"
+	NonTerminalStateIndex = ".status.state.nonTerminalWorker"
```

Comment:
Curious to see if it would make sense to break this out on a per-state basis instead of just non-terminal states.

I'm also wondering if we might run into issues based on the field naming here, given the comment about the `FieldIndexer` interface here: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.1/pkg/client#FieldIndexer

Reply:
> Curious to see if it would make sense to break this out on a per-state basis instead of just non-terminal states

I've added an additional index for state, `WorkflowStateIndex` and I've updated the index variable names for clarity. Take a look at the documented example under `FieldIndexer`, the Index name is just a key to lookup an object based on shared understanding of what that index means. In the documented example, the field name is `spec.volumes.secret.secretName` and returns all the secret names mounted on a pod.